### ISSUE TITLE
Codecov concern

### ISSUE THREAD

**Issue by taiho-nordstrom** [apollographql/apollo-tooling#2304]
Hello Apollo team,

In response to a[ recent breach of Codecov](https://about.codecov.io/security-update/), my team did a quick research and found out that one of the dependencies in ApolloCLI does use that affected command - `bash <(curl -s https://codecov.io/bash)`

```bash
$ grep "https://codecov.io/bash" -R .
./Codegen/ApolloCLI/apollo/node_modules/es-abstract/.travis.yml:  - 'if [ -n "${COVERAGE-}" ]; then npm run coverage && bash <(curl -s https://codecov.io/bash) -f coverage/*.json; fi'
```

I am surprised that the Travis CI config file was included in the release bundles. Is there anything your team can do to not expose the yml file that has that affected command

**Comment by abernix** [comment]
Thanks for raising your concern.  I've reviewed the disclosure notice you've linked to.

You've provided a `grep` from your codebase, but I'm also not sure how to reproduce the installation you're demonstrating with any version of `apollo` published after April 17, 2020 since that's the version that https://github.com/apollographql/apollo-tooling/pull/1883 landed in and would have brought in a version of `es-abstract` older than `^1.18.0-next.1`, which was a minimum constraint for `es-abstract` its dependency on `object.getownpropertydescriptors`.  I don't believe this would happen with any recent install of `apollo`.  Can you provide the output of `npm ls apollo` in your project? Have you updated your `apollo` to the latest version?

Even on an older version, your assessment that we **use** the affected command also seems incorrect to me.  You are correct that there is a _file_ present in your _current_ installation of `apollo` that _has_ the once-affected command in it, but the implication that it is **used** in any doesn't seem validated to me.

The `.travis.yml` you're seeing present in your installation (based on your command output) is not from this project but from a transitive dependency_: [the `es-abstract` module](https://www.npmjs.com/package/es-abstract) which is a dependency of `object.getownpropertydescriptors` which is a dependency of `util.promisify` which is a dependency of `git-parse`.  This is what the tree looks like:

```
.
└─┬ apollo
  └─┬ git-parse
    └─┬ util.promisify
      └─┬ object.getownpropertydescriptors
        └── es-abstract
```


It appears that `es-abstract` was indeed not excluding `.travis.yml` _in their npm publishing process_ and therefore the file was present in its published bundle up until version `1.17.7`.  The `es-abstract` project published a new version 2 months ago, `v1.18.0`, [which no longer has the `.travis.yml` in it](https://www.runpkg.com/?es-abstract@1.18.0/index.js).  (You can see it [present in `v1.17.7` here](https://www.runpkg.com/?es-abstract@1.18.0/index.js)).

That said, I don't think there is a requirement for them to not publish their `.travis.yml` file and I don't see how the mere presence of that file in a nested package could alone be exploited.  It _could_ relevant if:

1. Codecov's script was still exploited (Which Codecov has fixed; the script is not _still_ exploited)
2. Your project was invoking nested `.travis.yml` files.  This would be an unexpected situation / dramatic misconfiguration because Travis runs the `.travis.yml` at root of projects or your project was blindly running `.travis.yml` files through some sort of interpreter that understands that format and executes its commands in shell — also pretty unexpected.

This is certainly not the case for `apollo` or this `apollo-tooling` repository and this doesn't seem particularly likely for any project.

All in all, I haven't seen any evidence to believe this is or even was exploitable by its mere presence in a dependency tree.  Perhaps on the `es-abstract` project itself (meaning _its own environment variables_ might have been vulnerable) but that file hasn't been present on the repository itself for many months (back to 2020, long before this exploit), and certainly not anymore.

Just for good measure, despite it appearing completely not vulnerable to me, since it does appear that we still have some artifacts of an older `es-abstract` package in our monorepo's `package-lock.json`, I've gone ahead and updated it with #2307.  I will publish a new version momentarily, **just to be sure.**

**Comment by taiho-nordstrom** [comment]
I totally agreed with you. Our project was not directly affected by this breach, but nevertheless, it's totally not a good thing to see in our codebase.

We use Apollo iOS SDK v0.26.0. Here is its pin in the Package.resolved file
```json
 {
        "package": "Apollo",
        "repositoryURL": "https://github.com/apollographql/apollo-ios.git",
        "state": {
          "branch": null,
          "revision": "cf3d35cdf64bdb07f089eddc7df2584800d46fb0",
          "version": "0.26.0"
        }
}
```

I was a bit surprised to see a node package in an iOS project. I am not sure how ApolloCLI uses it though.

**Comment by abernix** [comment]
The `apollo` CLI is used for its Swift codegen.

As I noted I would above, I've published `apollo@2.33.1` with https://github.com/apollographql/apollo-tooling/pull/2307.  That would need updates to land in a current Apollo iOS version.

I'm not perfectly square on timing, but the update to `apollo` in iOS that I would have expected to update this dependency in your stack happened just around the time that Apollo iOS 0.26.0 came out.  Not sure how much you can upgrade, but 0.26 is pretty far from recent releases.  You might have success eradicating your concern with even just https://github.com/apollographql/apollo-ios/releases/tag/0.29.0, but again, I don't think it's actually worrisome, even if it was prudent of you to check into.

I'm going to close this, but @designatednerd  might be able to help dissect version overlaps here more if you need it.

**Comment by taiho-nordstrom** [comment]
Thank you for looking into this!
